### PR TITLE
Calendar sync from an api fetch

### PR DESCRIPTION
**Pull request by colemanmoore**

### CODE REVIEW

**Comment by colemanmoore** [comment]
@solomon-b This seems to be fixing the issue - the calendar reflects new, edited, and removed events, but it takes about 5 minutes to update for whatever reason. Note that I also made a change to the archive api route, but it is working on the preview the same as in production

Another round of refactoring is in order to make all the pages and api routes consistent with the v14 patterns specified in the nextjs docs, which may not have been the key fix here (the SWR hook ultimately solved it), but would be good to get up to date, the docs were hard to follow with the different project structure

**Comment by solomon-b** [comment]
Thanks so much @colemanmoore!

**Review by solomon-b — APPROVED** [review]
